### DISCUSSION TITLE
Add ability to hide columns in Table Editor

### DISCUSSION THREAD

**Discussion by sunech** [supabase/supabase#18197] [Feature Requests]
Would be nice to add a Columns button, after the Refresh/Filter/Sort buttons in Table Editor.

Ideally it should present a list of columns with checkboxes. Unchecking a column, should hide it from the table editor.

This would enable users working on large tables to customize the view, to hide columns that are not relevant. It would increase productivity and reduce risk of mistakes.

Making it possible to drag/drop the columns in the same list to reorder them, would be a plus as well.

**Comment by sunech** [comment]
I just figured out Prisma Studio has this feature already, something similar to their implementation would be great:
![image](https://github.com/supabase/supabase/assets/93080204/bc2abdb2-936e-4c13-bc4d-c1cecaa8fb1b)

**Comment by chamals3n4** [comment]
yeah, i'm also looking for this feature. this is how i make a view currently

```sql
CREATE VIEW custom_view AS
SELECT col1, col2, col3
FROM table_name;
